pkg/hub

```diff
@@ -0,0 +1,21 @@
+[mirrors]
+https://github.com/github/hub/archive/v2.5.1.tar.gz
+
+[vars]
+filesize=990972
+sha512=31a9c721dd29e9a7aee28bccb36621802c6d85dbb7277a2f55365c2fcaec4f4a587e6eb7a6fd9231c7c86acdf13ccad9c7846e12ad6f76812943e8d5eb0d9383
+tarball=hub-2.5.1.tar.gz
+tardir=hub-2.5.1
+desc='git wrapper for using github (fork, pr, etc) on the commmand line'
+pkgver=1
+
+[deps]
+go
+git
+
+[build]
+mkdir -p "$butch_build_root"/src
```

Comment:
i find it kinda confusing to see butch_build_root in use, when it's practically equivalent to "." inside recipes (unless cd was used).
apart from that this looks like a really nice trick :)

Reply:
hmm on second glance i see butch_build_root is actually /src/build/packagename and butch_build_dir /src/build/packagename/tardir.
the buildroot dir is never cleaned up, so it's probably not a good idea to install stuff into it.

would it work like this?
```
mkdir -p src/packagename
GOPATH=src \
 ...

```
maybe you could clarify the idea behind the symlink / gopath trick ?

Reply:
go expects a directory tree like this GOPATH/{bin,src,pkg} and any directories under src with .go files in them are valid paths to give to `go cmd`. ie we link the source files of the tarball into GOPATH/src/pkg_name and now `go install pkg_name` can be used. GOBIN dictates where the binaries will be installed. By default GOBIN is GOPATH/bin. I put mkdir "$butch_build_root"/src instead of mkdir -p src/ because I did not want to overwrite a directory named src in a tarball. I am not very familiar with golang so I don't know if that's a real concern. There would also be a recursive link as butch_build_dir needs to be linked into the src/pkg_name path.

Reply:
maybe the script should clean up after itself? ie rm build_root/src/pkg_name?  or perhaps use /tmp instead

Reply:
hmm, how about __sabotage_go_builddir__/src ? or something like that

Reply:
where `sabotage_go_builddir=$(mktemp -d)` or what? I'm not sure where to put it. I like the current solution, as it's only a directory and link that's made. If you have another idea, I'll happily hack something up.

Reply:
no, i mean inside the current builddir. i think it's much nicer if you have temporary build stuff in the build dir...
so in the recipe head:
```
[build]
mkdir -p sabotage_temp/src
ln -sf ../.. sabotage_temp/src/$butch_package_name
```

Reply:
oh not bad. I went with the your first recommendation of doing `mkdir -p src/etc`. I figure if a build breaks because a directory is named src, then we deal with it then. I made the changes before I saw this comment - I'll switch it.

Reply:
`mkdir -p butch_build_tmp/etc`
I had to change the build to include the url because I made a mistake when testing and rebuilding earlier.

Reply:
hmm why do you keep on using $butch_build_dir ? i think just using "." or "../.." for the symlink is much cleaner
- a reader knows immediately what "." or ".." refers to
- it's much shorter